eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/ImmutableMapFactoryImpl.java

```diff
@@ -132,23 +133,42 @@
     }
 
     /**
-     * @deprecated use {@link #ofAll(Map)} instead (inlineable)
+     * @deprecated use {@link #withMap(Map)} instead (inlineable).
      */
     @Override
     @Deprecated
     public <K, V> ImmutableMap<K, V> ofMap(Map<K, V> map)
     {
-        return this.ofAll(map);
+        return this.withMap(map);
     }
 
+    /**
+     * Same as {@link #withMap(Map)}.
+     * @deprecated since 10.3.0, use {@link #withMap(Map)} instead.
+     */
     @Override
+    @Deprecated
     public <K, V> ImmutableMap<K, V> ofAll(Map<K, V> map)
     {
-        return this.withAll(map);
+        return this.withMap(map);
     }
 
+    /**
+     * Same as {@link #withMap(Map)}.
+     * @deprecated since 10.3.0, use {@link #withMap(Map)} instead.
+     */
     @Override
+    @Deprecated
     public <K, V> ImmutableMap<K, V> withAll(Map<K, V> map)
+    {
+        return this.withMap(map);
+    }
+
+    /**
+     * @since 10.3.0
+     */
+    @Override
+    public <K, V> ImmutableMap<K, V> withMap(Map<K, V> map)
```

Comment:
I feel like the change you're making is different than what the title says. You're really adding a new method called withMap and deprecating ofAll and withAll at the same time in favor of it.

Reply:
In addition, this only changes ImmutableMapFactory. It misses other object maps (BiMap, hashing strategy) and primitive maps.

Reply:
> I feel like the change you're making is different than what the title says. You're really adding a new method called withMap and deprecating ofAll and withAll at the same time in favor of it.

@motlin I kept phrasing of the title similar to the original issue gh-738.